### PR TITLE
frontend/enhancement/coin-nudges

### PR DESCRIPTION
**Pull request by DwivediAman-exe**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Flow.purs [hunk 1]
```diff
@@ -3400,10 +3411,53 @@ updateBannerAndPopupFlags = do
...
+    ST.ONE_MORE_RIDE ->
+      if isPopupShownToday coinPopupInfo.oneMoreRide && checkCoinIsEnabled && appConfig.coinsConfig.eightRideCoinEvent
+        then ST.ONE_MORE_RIDE
+        else ST.NO_COIN_POPUP
+    ST.TWO_MORE_RIDES ->
+      if isPopupShownToday coinPopupInfo.twoMoreRides && checkCoinIsEnabled && appConfig.coinsConfig.eightRideCoinEvent
+        then ST.TWO_MORE_RIDES
+        else ST.NO_COIN_POPUP
+    ST.REFER_AND_EARN_COIN ->
+      if isPopupShownToday coinPopupInfo.referAndEarnCoin && withinTimeRange "07:30:00" "23:59:59" (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss") && checkCoinIsEnabled && appConfig.coinsConfig.driverToCustomerRefCoinEvent && (not isDateGreaterThan appConfig.coinsConfig.driverToCustomerRefPopupEndDate)
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Flow.purs:3450]: store the time in constant.purs

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs [hunk 2]
```diff
@@ -2008,3 +2008,132 @@ introducingCoinsPopup state = PopUpModal.config {
...
+        { text = popupConfig.option1
+        , color = Color.yellow900
+        , background = Color.black900
+        , visibility = true
+        , margin = MarginTop 16
+        , width = MATCH_PARENT
+        }
+      , coverImageConfig
+        { imageUrl = popupConfig.coverImageConfig
+        , visibility = if popupConfig.coverImageConfig == "" then GONE else VISIBLE
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:2042]: use bool to visibility

**Review by MercyQueen — APPROVED** [review]

**Review by Vignesh-772 — COMMENTED** [review]
Resolve the comments

File: Frontend/ui-driver/src/Services/Backend.purs [hunk 3]
```diff
@@ -1493,3 +1492,10 @@ mkUpdateDriverVehiclesServiceTier ridePreferences =
             tiers : tierArray,
             airConditioned : Nothing
         }
+
+getRideStatusPastDays :: String -> Flow GlobalState (Either ErrorResponse RideStatusPastDaysRes)
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Services/Backend.purs:1496]: bro use callApi frunctions from `Helpers.API` instead

**Review by witcher-shailesh — DISMISSED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**